private/buf/bufgen/generator.go

```diff
@@ -169,15 +169,25 @@ func (g *generator) execPlugins(
 	jobs := make([]func(context.Context) error, 0, len(config.PluginConfigs))
 	responses := make([]*pluginpb.CodeGeneratorResponse, len(config.PluginConfigs))
 	requiredFeatures := computeRequiredFeatures(image)
+	remotePluginConfigTable := make(map[string]map[int]*PluginConfig, len(config.PluginConfigs))
```

Comment:
Instead of two layers of maps, can we have a map of a slice of structs with something like:

```go
type remotePluginExec struct {
  ResponseIndex int
  Config *PluginConfig
}
```

Then for debugging, it is a little easier to figure out what the `int` is used for and we don't shuffle around the remote plugins to execute unnecessarily. I know order doesn't matter in plugin execution until we assemble the results at the end, but it would be nice for debugging if the order specified in the buf.gen.yaml file was kept.